scripts/actions/abilities/pets/nightmare.lua

```diff
@@ -0,0 +1,45 @@
+-----------------------------------
+-- Nightmare - Player's Avatar
+-- AoE Sleep
+-- Sleep that is not broken from DoT effects (any dmg source that doesn't break bind).
+-- This version of it is from a player's avatar. The sleep is broken by most damage sources except other DoTs
+--
+-- see mobskills/nightmare.lua for full explanation
+-----------------------------------
+local abilityObject = {}
+
+abilityObject.onAbilityCheck = function(player, target, ability)
+    return 0, 0
+end
+
+abilityObject.onPetAbility = function(target, pet, skill)
+    local duration = 90
+    local dotdamage = 2
+    local sleepTier = 1
+    local dINT = pet:getStat(xi.mod.INT) - target:getStat(xi.mod.INT)
+    local bonus = xi.summon.getSummoningSkillOverCap(pet)
+    local resm = xi.mobskills.applyPlayerResistance(pet, -1, target, dINT, bonus, xi.element.DARK)
+    if resm < 0.5 then
+        skill:setMsg(xi.msg.basic.JA_MISS_2) -- resist message
+        return xi.effect.SLEEP_I
+    end
+
+    duration = duration * resm
+    if
+        target:hasImmunity(1) or
```

Comment:
`1`? is there no enum for this?

Reply:
one exists in [battleentity.h](#L457) but becaus eit hasn't been added in scripts/enum yet there are both magic numbers as well commented out lua that probably shoudln't even exists https://github.com/search?q=repo%3ALandSandBoat%2Fserver+immunity+language%3ALua&type=code&l=Lua because those mobs pools should be doing the deed 

tl;dr we need to add a new file to the enum folder or start coming up with methods that share C++ enums to lau or lua enums to C++ so we stop having things in 2 places

Reply:
We can do a pass to fix immunity enum later